_posts/en/meetings/2018-06-21-meeting.md

```diff
@@ -0,0 +1,336 @@
+valid, and other migration concerns.
+
+**Conclusion:** Bishop will send an email to the mailing list, hopefully
+before migration to a new host domain, with additional details once he
+has them.
+
+## Coin selection
+
+**Background:** several developers have been working on improving
+Bitcoin Core's coin selection---how it chooses which bitcoins (inputs)
+to spend---to simultaneously improve privacy, reduce transaction size,
+and reduce fees.  The current selection protocol starts with a
+Branch-and-Bound (BnB) algorithm that tries to find a match between the
+inputs available and the amount being sent.  If that doesn't work, a
+fallback algorithm is needed.  A Single-Random-Draw (SRD) algorithm
+randomly adds additional inputs to a partial transaction until the sum
+of the inputs is equal to or greater than the amount being spent
+(including fees).
+
+This week's discussion is a continuation of [last week's discussion][srd
+2018-06-14] about the same topic.
+
+**Discussion ([log][log coin selection]):** Andrew Chow requested and
+introduced the topic, "I did a bunch of simulations of the
+[single random draw] fallback stuff
+([link](https://gist.github.com/achow101/242470486265d3f21adab08f65b9102c)).
+There are two problems that I see with this strategy: change can be
+incredibly small and the mean number of UTXOs in the wallet is quite
+high.  The question is whether we can accept these tradeoffs or whether
+we need to find a better algorithm."
+
+Gregory Maxwell said, "[If I recall correctly], there is nothing
+fundamental about [single random draw] that makes it good for making
+[branch and bound] work better, but rater it was the first alternative
+\[Mark Erhardt] tried there."
+
+Chow added, "well, and in [Erhardt]'s simulations, [single random
+draw] performed reasonably well and was extremely simple.  Though I
+guess we may be seeing different results now."
+
+Various additional strategies and their tradeoffs were discussed, but
+the topic was starting to become complicated for a short segment of a
+time-limited text-only meeting.
+
+**Conclusion:** Chow suggested, "perhaps this coin selection discussion
+would be better done in person with whiteboards," ending the meeting
+discussion, although Maxwell noted, "that leaves out people who can't
+attend."  Presumably discussion will continue on PR [#13307][] and
+perhaps elsewhere.
+
+## Multiwallet session persistence
+
+**Background:** the development branch ("master" branch) of Bitcoin Core
+includes code that allows users to dynamically load and unload
+individual wallets in multiwallet mode.  For example, you can have a
+"personal" wallet and a "business" wallet that can each be opened or closed
+separately.
+
+**Discussion ([log][log multiwallet persistence]):** Jonas Schnelli
+requested and introduced the topic, "I guess it's not ideal that loaded
+wallets need to be re-loaded after a Bitcoin Core restart---especially
+in pruning mode."
```

Comment:
This could perhaps use more context. The risk here is that someone:

- starts bitcoind
- loads a wallet at runtime
- stops bitcoind
- restarts bitcoind

At this point, the previously dynamically loaded wallet will not be loaded, the tip will advance, but the wallet will not be notified of new transactions. Therefore when the wallet is used again, it would require a rescan to ensure that it hasn't missed txs. Jonas's comment about pruning is that if the node has pruned beyond the wallet's last processed block in the meantime, it's not possible to do a rescan.